crates/kumod/src/smtp_server.rs

```diff
@@ -853,13 +911,14 @@ impl SmtpServerSession {
         let mut meta = ConnectionMetaData::new();
         meta.set_meta("reception_protocol", "ESMTP");
         meta.set_meta("received_via", my_address.to_string());
-        meta.set_meta("received_from", peer_address.to_string());
+        meta.set_meta("client_address", client_address.to_string());
```

Comment:
this breaks the `received_from` metadata and semantics; I think we should preserve this how it was.

Instead, when we handle the proxy protocol, we should capture the `peer_address` into a `proxy_address` (or better name) meta item, then update `received_from` to be the claimed peer address behind the proxy, as well as a `proxy_mode` (or better name) meta item describing what kind of proxy is in use.

I think that results in maximal sanity!